common/workunit/workunit.cpp

```diff
@@ -9492,9 +9492,25 @@ IPropertyTree *CLocalWorkUnit::getUnpackedTree(bool includeProgress) const
     return ret.getClear();
 }
 
+IPropertyTree *CLocalWorkUnit::queryIndirectTree() const
+{
+    if (indirectTree)
+        return indirectTree;
+    if (!p->hasProp("@clonedFromWorkunit"))
+        return p;
+    else
+    {
+        indirectTree.setown(createPTreeFromIPT(p));
+        Owned<IWorkUnitFactory> factory = getWorkUnitFactory();
+        Owned<IConstWorkUnit> donor = factory->openWorkUnit(p->queryProp("@clonedFromWorkunit"));
+        copyTree(indirectTree, queryExtendedWU(donor)->queryPTree(), "Graphs");
```

Comment:
Should check if the workunit does not exist?

Reply:
Yes, probably